File: core/chunking.py
from .constants import CHUNK_SIZE


def split_into_chunks(data: bytes, chunk_size: int = CHUNK_SIZE) -> list[bytes]:
    if not data:
        return [data]
    return [data[i : i + chunk_size] for i in range(0, len(data), chunk_size)]


def join_chunks(chunks: list[bytes]) -> bytes:
    return b"".join(chunks)

File: webapp/chunking.js
// Mirrors core/chunking.py::split_into_chunks -- pure Uint8Array slicing, always
// emitting at least one chunk (even for empty input), no padding logic needed (base64
// already round-trips a partial final chunk exactly).
import { CHUNK_SIZE } from './constants.js';

export function splitIntoChunks(data, chunkSize = CHUNK_SIZE) {
  if (data.length === 0) {
    return [data];
  }
  const chunks = [];
  for (let i = 0; i < data.length; i += chunkSize) {
    chunks.push(data.slice(i, i + chunkSize));
  }
  return chunks;
}
